File: app/src/main/java/com/movetoplay/screens/SignInActivity.java
package com.movetoplay.screens;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.movetoplay.R;
import com.movetoplay.screens.applock.LimitationAppActivity;

public class SignInActivity extends AppCompatActivity {

    private EditText etEmail, etPassword;
    private CheckBox checkStatus;
    private Button btnEnter;
    private TextView tvForgotPassword;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_in);

        initViews();
        initListeners();
    }

    @Override
    public void onBackPressed() {
        finish();
        super.onBackPressed();
    }

    private void initListeners() {
        btnEnter.setOnClickListener(view -> {
            String email = etEmail.getText().toString();
            String password = etPassword.getText().toString();

            if (!email.isEmpty() && !password.isEmpty()) {
                if (checkStatus.isChecked()) {
                    startActivity(new Intent(getApplicationContext(), SetupProfileActivity.class));
                } else {
                    startActivity(new Intent(getApplicationContext(), LimitationAppActivity.class));
                }
                finish();
            } else {
                Toast.makeText(this, "Заполните все поля", Toast.LENGTH_SHORT).show();
            }
        });

        tvForgotPassword.setOnClickListener(v -> {
            startActivity(new Intent(getApplicationContext(), ResetPass.class));
            finish();
        });
    }

    private void initViews() {
        etEmail = findViewById(R.id.email);
        etPassword = findViewById(R.id.password);
        checkStatus = findViewById(R.id.checkBox);
        btnEnter = findViewById(R.id.btn_enter);
        tvForgotPassword = findViewById(R.id.tv_forgot_password);

    }
}

File: app/src/main/java/com/movetoplay/screens/SignInActivity.kt
package com.movetoplay.screens

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.movetoplay.databinding.ActivitySignInBinding
import com.movetoplay.pref.Pref
import com.movetoplay.screens.applock.LimitationAppActivity

class SignInActivity: AppCompatActivity() {
    private lateinit var binding: ActivitySignInBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySignInBinding.inflate(layoutInflater)
        setContentView(binding.root)
        initListeners()
    }

    override fun onBackPressed() {
        finish()
        super.onBackPressed()
    }

    private fun initListeners(){
        val email: String = binding.email.text.toString()
        val password: String = binding.password.text.toString()
        Pref.userLogin = binding.email.text.toString()
        Log.e("Pref", Pref.userLogin)
        binding.btnEnter.setOnClickListener{
            Pref.userLogin = binding.email.text.toString()
            if (email.isBlank()||password.isBlank()){
                if (binding.checkBox.isChecked){
                    Pref.userLogin = binding.email.text.toString()
                    startActivity(
                        Intent(
                            applicationContext,
                            SetupProfileActivity::class.java
                        )
                    )
                } else {
                    Pref.parentsLogin = binding.email.text.toString()
                    Log.e("parentsLogin", Pref.parentsLogin)
                    startActivity(
                        Intent(
                            applicationContext,
                            LimitationAppActivity::class.java
                        )
                    )
                }
            } else {
                Toast.makeText(this, "Заполните все поля", Toast.LENGTH_SHORT).show()
            }
        }
    }

//    fun isEntryValid( role: String, name: String, lastName: String, age: Int): Boolean {
//        if (role.isBlank()||name.isBlank() || lastName.isBlank() || age.toString().isBlank()) {
//            return false
//        }
//        return true
//    }
}